packages/widgetbook_cli/bin/command/publish.dart

```diff
@@ -0,0 +1,234 @@
+      'build',
+      'web',
+    );
+
+    final directory = Directory(buildPath);
+    final useCases = baseBranch == null
+        ? <ChangedUseCase>[]
+        : await UseCaseParser(
+            projectPath: path,
+            baseBranch: baseBranch,
+          ).parse();
+    final themes = await ThemeParser(projectPath: path).parse();
+    final locales = await LocaleParser(projectPath: path).parse();
+    final devices = await DeviceParser(projectPath: path).parse();
+    final textScaleFactors =
+        await TextScaleFactorParser(projectPath: path).parse();
+    final file = WidgetbookZipEncoder().encode(directory);
+    if (file != null) {
+      final uploadInfo = await WidgetbookHttpClient().uploadDeployment(
+        deploymentFile: file,
+        data: DeploymentData(
+          branchName: branch,
+          repositoryName: repository!,
+          commitSha: commit,
+          actor: actor!,
+          apiKey: apiKey,
+          provider: gitProvider,
+        ),
+      );
+
+      if (uploadInfo != null && prNumber != null) {
+        if (gitHubToken != null) {
+          await GithubProvider(
+            apiKey: gitHubToken,
+          ).addBuildComment(
+            buildInfo: uploadInfo,
+            number: prNumber,
+          );
+        }
+      }
+
+      // If generator is not run or not properly configured
+      if (themes.isEmpty) {
+        logger.err(
+          'HINT: Could not find generator files. '
+          'Therefore, no review has been created. '
+          'Make sure to use widgetbook_generator and '
+          'run build_runner before this CLI. '
+          'See https://docs.widgetbook.io/widgetbook-cloud/review for more '
+          'information.',
+        );
+        exit(0);
+      }
+
+      if (uploadInfo != null && baseBranch != null && baseCommit != null) {
+        await WidgetbookHttpClient().uploadReview(
+          apiKey: apiKey,
+          useCases: useCases,
+          buildId: uploadInfo['build'] as String,
+          projectId: uploadInfo['project'] as String,
+          baseBranch: baseBranch,
+          baseSha: baseCommit,
+          headBranch: branch,
+          headSha: commit,
+          themes: themes,
+          locales: locales,
+          devices: devices,
+          textScaleFactors: textScaleFactors,
+        );
+      } else {
+        logger.warn(
+          'HINT: No pull-request information available. Therefore, '
+          'no review will be created. See docs for more information.',
+        );
+      }
+    } else {
+      logger.err('Could not create .zip file for upload.');
+    }
+    return ExitCode.success.code;
+  }
+
+  Future<String> _getActorName() async {
+    final results = await Process.run('git', ['config', 'user.name']);
+    return results.stdout as String;
+  }
+
+  Future<String> _getRepositoryName() async {
+    final results = await Process.run('git', ['rev-parse', '--show-toplevel']);
+    return results.stdout.toString().split('/').last;
```

Comment:
should use git package

Reply:
Yes already solved this .. now using widgetbook_git